sbi/inference/snl/snl.py

```diff
@@ -142,8 +142,10 @@ def __call__(
             # therefore return a theta vector with the same ordering as x.
             theta, x = self._batched_simulator(theta)
             # Store (theta, x) pairs.
-            self._theta_bank.append(theta)
-            self._x_bank.append(x)
+            # Get indices with finite data to remove NaN simulations.
+            x_not_nan = torch.unique(torch.where(torch.isfinite(x))[0])
```

Comment:
I am not sure about unique; it's not documented. 
In fact, I don't understand the expression at all. First `isfinite` produces a bool array. Then you get the `torch.nonzero` (that's what `where(..)[0]` does indexes. Indexes by definition cannot repeat. So what's the point of unique?

Have you thought through how this shuold work in several dimensions? 

Overall, this statement repeats 3 times, is not well-documented and has potential for bugs. It deserves its own function, perhaps in `user_input_validation` with some doctests showing how it works on several paradigmatic `x`.

Reply:
Also something to consider is whether we will silently discard a lot of NaN input without reporting and the user will get a posterior based on very little data but be under the impression that it is based on a great deal of data.